libfwupd/fwupd-device.c

```diff
@@ -1662,6 +1664,107 @@ fwupd_device_has_flag(FwupdDevice *self, FwupdDeviceFlags flag)
 	return (priv->flags & flag) > 0;
 }
 
+/**
+ * fwupd_device_get_problems:
+ * @self: a #FwupdDevice
+ *
+ * Gets device problems.
+ *
+ * Returns: device problems, or 0 if unset
+ *
+ * Since: 1.8.1
+ **/
+guint64
+fwupd_device_get_problems(FwupdDevice *self)
+{
+	FwupdDevicePrivate *priv = GET_PRIVATE(self);
+	g_return_val_if_fail(FWUPD_IS_DEVICE(self), 0);
+	return priv->problems;
+}
+
+/**
+ * fwupd_device_set_problems:
+ * @self: a #FwupdDevice
+ * @problems: device problems, e.g. %FWUPD_DEVICE_PROBLEM_SYSTEM_POWER_TOO_LOW
+ *
+ * Sets device problems.
+ *
+ * Since: 1.8.1
+ **/
+void
+fwupd_device_set_problems(FwupdDevice *self, guint64 problems)
+{
+	FwupdDevicePrivate *priv = GET_PRIVATE(self);
+	g_return_if_fail(FWUPD_IS_DEVICE(self));
+	if (priv->problems == problems)
+		return;
+	priv->problems = problems;
+	g_object_notify(G_OBJECT(self), "problems");
+}
+
+/**
+ * fwupd_device_add_problem:
+ * @self: a #FwupdDevice
+ * @problem: the #FwupdDeviceProblem, e.g. #FWUPD_DEVICE_PROBLEM_SYSTEM_POWER_TOO_LOW
+ *
+ * Adds a specific device problem kind to the device.
+ *
+ * Since: 1.8.1
+ **/
+void
+fwupd_device_add_problem(FwupdDevice *self, FwupdDeviceProblem problem)
+{
+	FwupdDevicePrivate *priv = GET_PRIVATE(self);
+	g_return_if_fail(FWUPD_IS_DEVICE(self));
+	if (problem == 0)
+		return;
+	if ((priv->problems | problem) == priv->problems)
```

Comment:
Shouldn't this be &

Reply:
No, as it's comparing it with itself, but it's a weird construct, and we can just use fwupd_device_has_problem instead....